packages/react-log-viewer/src/LogViewer/LogViewer.tsx

```diff
@@ -0,0 +1,157 @@
+  // else, create new canvas
+  canvas = canvas || document.createElement('canvas');
+  const context = canvas.getContext('2d');
+  context.font = font;
+  const metrics = context.measureText(text);
+  return metrics.width;
+}
+
+export const LogViewer: React.FunctionComponent<LogViewerProps> = memo(
+  ({
+    data = '',
+    hasToolbar = true,
+    customControls = null,
+    placeholder = 'Search',
+    hasLineNumbers = true,
+    width = 800
+  }: LogViewerProps) => {
+    const [searchedInput, setSearchedInput] = useState<string | null>('');
+    const [rowInFocus, setRowInFocus] = useState<number | null>(null);
+    const [searchedWordIndexes, setSearchedWordIndexes] = useState<number[] | null>([]);
+    const [highlightedRowIndexes, setHighlightedRowIndexes] = useState<number[] | null>([]);
+    const [currentSearchedItemCount, setCurrentSearchedItemCount] = useState<number>(0);
+    const [parsedData, setParsedData] = useState<string[] | null>([]);
+    const loggerRef = React.createRef<any>();
+
+    const dataToRender = React.useMemo(
+      () => ({
+        parsedData,
+        loggerRef,
+        rowInFocus,
+        searchedWordIndexes,
+        highlightedRowIndexes,
+        setHighlightedRowIndexes
+      }),
+      [parsedData, loggerRef, rowInFocus, searchedWordIndexes, highlightedRowIndexes, setHighlightedRowIndexes]
+    );
+
+    /* Parsing our data immediately after first component render */
+    useEffect(() => {
+      setParsedData(parseConsoleOutput(data));
+    }, [data]);
+
+    /* Updating searchedResults context state given changes in searched input */
+    useEffect(() => {
+      let foundKeywordIndexes: (number | null)[] = [];
+      const adjustedSearchedInput = escapeString(searchedInput);
+
+      if (adjustedSearchedInput !== '' && adjustedSearchedInput.length >= 3) {
+        foundKeywordIndexes = searchForKeyword(adjustedSearchedInput, parsedData);
+
+        if (foundKeywordIndexes.length !== 0) {
+          setSearchedWordIndexes(foundKeywordIndexes);
+          scrollToRow(foundKeywordIndexes[DEFAULT_SEARCH_INDEX]);
+        }
+      }
+
+      if (adjustedSearchedInput !== '' && adjustedSearchedInput.length < 3) {
+        setRowInFocus(DEFAULT_FOCUS);
+        setCurrentSearchedItemCount(DEFAULT_INDEX);
+        setSearchedWordIndexes([]);
+      }
+
+      if (adjustedSearchedInput === '') {
+        setRowInFocus(DEFAULT_FOCUS);
+        // setCurrentSearchedItemCount(DEFAULT_INDEX);
+        // setSearchedWordIndexes([]);
+      }
+    }, [searchedInput]);
+
+    const scrollToRow = (searchedRowIndex: number) => {
+      setRowInFocus(searchedRowIndex);
+      loggerRef.current.scrollToItem(searchedRowIndex, 'center');
+      return true;
+    };
+
+    const loggerWidth = loggerRef.current ? loggerRef.current.clientWidth : 800;
+    const guessRowHeight = (rowIndex: number) => {
+      const rowText = parsedData[rowIndex];
+      const textWidth = getTextWidth(rowText, 'Liberation Mono');
```

Comment:
@nicolethoen Do you think it's worth making the font a prop, in case styling overrides happen? Or possibly the whole guessRowHeight function.